src/cpp/include/SIO/SIOReader.h

```diff
@@ -51,56 +51,55 @@ class SIOEventHandler ;
      * user.
      * @throws IOException
      */
-    virtual void open(const std::vector<std::string>& filenames) 
-      throw (IO::IOException, std::exception) ;
+    virtual void open(const std::vector<std::string>& filenames);
 
 
     /** Opens a file for reading (read-only).
      * @throws IOException
```

Comment:
We should update the comments too, this no longer throws and IOException. There are a few places where the comments can be updated.

Reply:
Unless I am really wrong about C++ method declarations, the "throw()" in the method declaration, or the absence of it, does not actually influence whether the code throws an exception or not. It was "decorative", in telling the compiler that this method might throw an exception, but never really taken into account by compilers.
If you are sure the code does not throw an IOException anywhere any longer, then yes, we can remove that comment.

Reply:
I am pretty sure this was a way to make an "interface", with the idea that these exceptions were thrown if these methods were not overloaded by the inheriting class but were called. This particular implementation of the class no longer throws the exception anymore and therefore should not claim to throw the exception and comments should be added where it is actually throw if ever thrown elsewhere.